### PR TITLE
ocamlPackages.fmt: 0.8.5 → 0.8.6

### PR DESCRIPTION
**Pull request by vbgl**
###### Motivation for this change

Needed to update alcotest

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/fmt/default.nix [hunk 1]
```diff
@@ -1,16 +1,20 @@
-{ stdenv, fetchurl, ocaml, findlib, ocamlbuild, topkg, cmdliner, result, uchar }:
+{ stdenv, fetchurl, ocaml, findlib, ocamlbuild, topkg, cmdliner, seq, stdlib-shims }:
+
+if !stdenv.lib.versionAtLeast ocaml.version "4.03"
+then throw "fmt is not available for OCaml ${ocaml.version}"
+else
 
 stdenv.mkDerivation {
-  name = "ocaml${ocaml.version}-fmt-0.8.5";
+  name = "ocaml${ocaml.version}-fmt-0.8.6";
```
Comment (bhipple) [inline pkgs/development/ocaml-modules/fmt/default.nix:8]:
```suggestion
  pname = "ocaml${ocaml.version}-fmt";
  version = "0.8.6";
```

File: pkgs/development/ocaml-modules/fmt/default.nix [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
 
   src = fetchurl {
-    url = https://erratique.ch/software/fmt/releases/fmt-0.8.5.tbz;
-    sha256 = "1zj9azcxcn6skmb69ykgmi9z8c50yskwg03wqgh87lypgjdcz060";
+    url = https://erratique.ch/software/fmt/releases/fmt-0.8.6.tbz;
```
Comment (bhipple) [inline pkgs/development/ocaml-modules/fmt/default.nix:11]:
```suggestion
    url = "https://erratique.ch/software/fmt/releases/fmt-${version}.tbz;"
```

File: pkgs/development/ocaml-modules/ppx_blob/default.nix [hunk 3]
```diff
@@ -9,10 +9,11 @@ buildDunePackage rec {
...
 
-  buildInputs = [ alcotest ocaml-migrate-parsetree ];
-  doCheck = true;
+  checkInputs = lib.optional doCheck alcotest;
+  buildInputs = [ ocaml-migrate-parsetree ];
+  doCheck = lib.versionAtLeast ocaml.version "4.03";
 
-  meta = with stdenv.lib; {
+  meta = with lib; {
     homepage = https://github.com/johnwhitington/ppx_blob;
```
Comment (bhipple) [inline pkgs/development/ocaml-modules/ppx_blob/default.nix:17]: Per https://github.com/NixOS/rfcs/pull/45, unquoted URLs are deprecated, so we're slowly cleaning them up as we go.  (Same for the `src` URL above.)

**Review by bhipple — CHANGES_REQUESTED** [review]

**Comment by vbgl** [comment]
Thanks for our comments. I’ve applied your suggestions.

**Comment by bhipple** [comment]
Result of `nixpkgs-review pr 84883` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
<br>- acgtk
<br>- jackline
<br>- stog
</details>

**Review by bhipple — APPROVED** [review]

**Pull request merged**